### PR TITLE
scala 2.12.0-M5 (devel)

### PR DESCRIPTION
**Pull request by cheleb**
- [x ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ?] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/scala.rb [hunk 1]
```diff
@@ -27,22 +27,22 @@ class Scala < Formula
...
 
     depends_on :java => "1.8+"
 
     resource "docs" do
-      url "http://www.scala-lang.org/files/archive/scala-docs-2.12.0-M4.zip"
-      sha256 "ccfbc72c5bfd3743b08f4ecfb3dea46cb2b982eaaba5ada24378a93c351f8c77"
-      version "2.12.0-M4"
+      url "http://www.scala-lang.org/files/archive/scala-docs-2.12.0-M5.zip"
+      sha256 "025d4db9755ffe0afc7254a105f95ca0ea6a77385d5c9b51a441aa8f34aefa5e"
+      version "2.12.0-M5"
```
Comment (ilovezfs) [inline Formula/scala.rb]: @DomT4 resource blocks can have versions? I'm confused :)

Comment (DomT4) [inline Formula/scala.rb]: Yeah, they can. Unnecessary in most cases, but yeah, they don't automatically inherit their parent's version.

Comment (ilovezfs) [inline Formula/scala.rb]: Is that ever used anywhere?

Comment (DomT4) [inline Formula/scala.rb]: Resource versioning? Mostly for cache sanity, I believe.

Comment (ilovezfs) [inline Formula/scala.rb]: I'm still not convinced this isn't an outright bug.

Comment (DomT4) [inline Formula/scala.rb]: Most of our versioning system is an outright bug :trollface:. Let's get this merged so we don't spam OP with brew implementation details, but agree at some point we need to handle `-` separated versions more sanely.

Comment (ilovezfs) [inline Formula/scala.rb]: What I'm saying is that I think OP could delete those versions from the formula and nothing will break.

Comment (DomT4) [inline Formula/scala.rb]: The version changes to `scala--docs-4.zip`, etc, which means we have resources in the cache that don't align with the expected devel version. That has the potential to cause issues when trying to cleanup the cache, etc.

Comment (ilovezfs) [inline Formula/scala.rb]: Wow, we don't propagate the explicit version from the spec? That's a pretty awesome cache bug. I'll open an issue in brew repo.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/f73de34c442fa6c599453877f349d6907461f69c. Thank you for your contribution to Homebrew @cheleb; we appreciate it! :smiley_cat:

**Pull request merged**